src/main.cpp

```diff
@@ -2799,20 +2799,40 @@ static int64_t nTimeFlush = 0;
 static int64_t nTimeChainState = 0;
 static int64_t nTimePostConnect = 0;
 
+struct CompareBlockIndexByWork {
+    bool operator()(const CBlockIndex* a, const CBlockIndex* b) const {
+        return a->nChainWork < b->nChainWork;
```

Comment:
Since its only used in each ActivateBestChainStep call individually, there is only ever one thing per given height/total work, but the map came out of an older version of the code and is now useless...so I'm replacing it with a vector.